pkg/operator/starter.go

```diff
@@ -16,20 +17,33 @@ import (
 	operatorclientv1 "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1"
 	operatorinformers "github.com/openshift/client-go/operator/informers/externalversions"
 	"github.com/openshift/library-go/pkg/controller/controllercmd"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
 	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+	"github.com/openshift/library-go/pkg/operator/staticresourcecontroller"
 	"github.com/openshift/library-go/pkg/operator/status"
 	"github.com/openshift/library-go/pkg/operator/v1helpers"
 
 	configobservationcontroller "github.com/openshift/cluster-openshift-controller-manager-operator/pkg/operator/configobservation/configobservercontroller"
 	"github.com/openshift/cluster-openshift-controller-manager-operator/pkg/operator/usercaobservation"
+	"github.com/openshift/cluster-openshift-controller-manager-operator/pkg/operator/v311_00_assets"
 	"github.com/openshift/cluster-openshift-controller-manager-operator/pkg/util"
 )
 
 func RunOperator(ctx context.Context, controllerConfig *controllercmd.ControllerContext) error {
-	kubeClient, err := kubernetes.NewForConfig(controllerConfig.ProtoKubeConfig)
+	// Increase QPS and burst to avoid client-side rate limits when reconciling RBAC.
+	// TODO: improve use of listers in the static resource sync controller
+	highRateLimitProtoKubeConfig := rest.CopyConfig(controllerConfig.ProtoKubeConfig)
+	if highRateLimitProtoKubeConfig.QPS < 50 {
+		highRateLimitProtoKubeConfig.QPS = 50
+	}
+	if highRateLimitProtoKubeConfig.Burst < 100 {
+		highRateLimitProtoKubeConfig.Burst = 100
+	}
+	kubeClient, err := kubernetes.NewForConfig(highRateLimitProtoKubeConfig)
```

Comment:
The static resource sync controller unfortunately doesn't use listers when reconciling RBAC objects.
This is problematic for ocm because we reconcile a lot of roles/clusterroles and their role bindings.
Long term we should be the ones to improve this in library-go, but for the interim we can increase the QPS and reduce the instances where the sync controller is throttled.

Reply:
So you are specifically referring to https://github.com/openshift/library-go/blob/c537a5b2af0196a9a235ccecc0d05c2583899f3d/pkg/operator/resource/resourceapply/generic.go#L150 as an example @adambkaplan, correct ?

Reply:
If so, a commit specific link to the file in question might help us to quickly remember where to go if we ever get to this TODO

Reply:
I think if I update the comment to use `StaticResourceController` then we should be ok (it is initialized later in this method).

Reply:
sorry I'm not following how updating the comment to use `StaticResourceController` pertains to my question around confirming where the shortcoming in library-go is and ask to somehow point to the location in that code for future reference

Reply:
Updated the comments - hopefully things are a little bit clearer.